docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -739,121 +745,213 @@ Specifically, `index` is in such a format:
 
-Clustered index affects the key format of primary index. Fortunately, the prefix of key range is untouched.
+### Behaviors
 
-Global secondary index largely affect the placement rules of partitioned tables. The key range of one global secondary index is not successive, so if it's necessary to define placement rules on the index, multiple rules should be generated in the PD. But for now, there's no such scenario.
+#### Syntax for Restricted Access (Compliance Case)
 
-## Privilege management
+Assume that if the example is logically like the "Compliance: User data needs geographic split" case:
 
-Privilege management is quite straightforward:
+```sql
+CREATE TABLE users (
+	id INT NOT NULL auto_increment,
+	username VARCHAR(64) NOT NULL,
+	email VARCHAR(64) NOT NULL,
+	dateofbirth DATE NOT NULL,
+	country VARCHAR(10) NOT NULL,
+	PRIMARY KEY (id),
+	UNIQUE (username)
+) PARTITION BY LIST COLUMNS (country) (
+	PARTITION pEurope VALUES IN ('DE', 'FR', 'GB') PLACEMENT POLICY='europe',
+	PARTITION pOther VALUES IN ('US', 'CA', 'MX')
+);
+```
 
-* `ALTER` statement requires `Alter` privilege
-* `information_schema.placement_rules` and `SHOW PLACEMENT POLICY` only shows the placement rules on the objects that visible to the current user
-* `ADMIN SHOW DDL` requires `Super` privilege
+What does `SHOW CREATE PLACEMENT POLICY europe` look like?
+
+I assume that it is something like:
+
+```sql
+CREATE PLACEMENT POLICY europe CONSTRAINTS="+region=eu-west-1" RESTRICTED;
+```
+
+This specific semantic will be the hardest to implement because of the other dependencies in the server.
+
+#### PD Balancing Behavior
+
+In the case of `FOLLOWER_CONSTRAINTS="[+region=us-east-1,+region=us-east-2]" FOLLOWERS=3`, is PD deterministic in balancing between each of the regions? If not, is it possible to ask PD to have certain scheduling/balancing policies.
+
+Other examples of scheduling policies (versus scheduling constraints) have been proposed by [@nolouch](https://github.com/nolouch), which I will paraphrase here:
+
+* `CREATE PLACEMENT POLICY p0 PRIMARY REGION="us-east-1" REGIONS="us-east-1,us-east-2" SCHEDULE=LOCALITY_IN_PRIMARY_REGION();`
+* `CREATE PLACEMENT POLICY p1 PRIMARY REGION="us-east-1" REGIONS="us-east-1,us-east-2" SCHEDULE=LOCALITY_BY_ROW();`
+* `CREATE PLACEMENT POLICY p2 PRIMARY REGION="us-east-1" REGIONS="us-east-1,us-east-2" SCHEDULE=SURVIVE_REGION_FAILURE();`
+
+The name `SCHEDULE` can be discussed, but in cases where there are 4 followers, being able to declaratively specify if you prefer full recovery over quick quorum is useful.
+
+#### Should we have "built-in" placement policies?
 
-## Ecosystem tools
+Examples could be "default", "system" etc. This might be helpful if you want to change the default or increase the follower count on system metadata.
```

Comment:
Is`default` mean cluster replica setting? I think it's necessary, PD has a default placement rule once the cluster is created.  and I think we also need an easy way to config the `meta` range data, `system` table range data.

Reply:
`default` means what to do for cases where a placement policy has not been specified. Typically you would expect users only change the replica settings, since adjusting other properties might be risky for balance.

I could see it being useful in an example like "Optimization: Multi-tenancy / control of shared resources". You might want to say that certain tikv servers are reserved and not supposed to be used for placement unless there is a placement policy.

If you think it is useful, I will add it into the proposal.